includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -122,11 +132,10 @@ public function scheduled_subscription_payment( $amount, $renewal_order ) {
 			return;
 		}
 
-		$payment_information = new Payment_Information( '', $renewal_order, $token, Payment_Initiated_By::MERCHANT() );
+		$payment_information = new Payment_Information( '', $renewal_order, Payment_Type::RECURRING(), $token, Payment_Initiated_By::MERCHANT() );
```

Comment:
It looks like we need to set `must_save_payment_method` here too. We might want to move this into Payment_info, i.e. automattically set `save_payment_method` to true when setting payment type to `recurring`, but not as part of this PR.

```suggestion
		$payment_information = new Payment_Information( '', $renewal_order, Payment_Type::RECURRING(), $token, Payment_Initiated_By::MERCHANT() );
		$payment_information->must_save_payment_method();
```

Reply:
Actually, `must_save_payment_method` is only relevant on customer-initiated payments. In this case `scheduled_subscription_payment` relies on an existing saved payment method, saving it again wouldn't make sense. Do you agree?

Reply:
Kind of, but I'm afraid creating precedent here. It looks more stable if save payment method is always true for subs payments. Another argument would be we passed it as `true` via arguments before those changes.

Reply:
It used to be `true`, because the same flag was used for the metadata, but now we have `Payment_Type::RECURRING()` for that. The purpose of this PR is to create a clear separation between `$force_save` and the metadata here.

And I think it makes even less sense to have the method now, because it would actually make things more confusing: Why would you save the payment method if a saved one is already being used?

Reply:
Agree. Pleas keep your approach 🙂